compiler/rustc_lint/src/types/literal.rs

```diff
@@ -5,7 +5,7 @@ use rustc_middle::ty::Ty;
 use rustc_middle::ty::layout::IntegerExt;
 use rustc_middle::{bug, ty};
 use rustc_span::Span;
-use {rustc_ast as ast, rustc_attr_parsing as attr, rustc_hir as hir};
+use {rustc_ast as ast, rustc_attr_data_structures as attrs, rustc_hir as hir};
```

Comment:
I think changing this back to `attr` would be nicer (and would make the diff smaller)

Reply:
There already is some `rustc_attr_parsing as attr` in the compiler, so using `rustc_attr_data_structures as attr` elsewhere feels wrong to me. Maybe we should rebind `rustc_attr_parsing` to something else?

Besides, `rustc_attr_data_structures` is where the attribute**s** live, so keeping the plural sounds right to me. But I'm fine with changing it as long as things stay consistent.

Reply:
Note also that there are some modules named `attr` and imports like `use rustc_ast::attr;` in various crates, so a style choice to have `rustc_attr_data_structures as attr` will lead to conflicts as we migrate more attributes.